gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java

```diff
@@ -21,54 +23,146 @@
 import org.apache.hadoop.hive.ql.metadata.Table;
 
 import com.google.common.base.Optional;
-import com.google.common.base.Splitter;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import com.typesafe.config.Config;
-import com.typesafe.config.ConfigException;
 import com.typesafe.config.ConfigFactory;
 
 import gobblin.data.management.copy.hive.HiveDataset;
+import gobblin.data.management.copy.hive.HiveDatasetFinder;
 import gobblin.hive.HiveMetastoreClientPool;
 import gobblin.util.ConfigUtils;
 
+
 /**
- * A {@link HiveDataset} that can be converted to another {@link HiveDataset}
+ * <p>
+ * A {@link HiveDataset} that can be converted from one source format to several destination formats.
+ * This class holds the {@link ConversionConfig}s required for conversion into each
+ * destination format. The {@link ConversionConfig} for a destination format can be accessed by calling {@link #getConversionConfigForFormat(String)}.
+ * </p>
+ *
+ * <p>
+ * <b>Instantiation</b>
+ * <ul>
+ *  <li> The constructor takes in a dataset {@link Config} which MUST have a comma separated list of destination formats at key,
+ *  {@value #DESTINATION_CONVERSION_FORMATS_KEY}
+ *  <li> Conversion configuration for a format can be set by using this destination format as prefix.
+ *  <li> E.g. If {@value #DESTINATION_CONVERSION_FORMATS_KEY}=flattenedOrc,nestedOrc.<br>
+ *  The destination table name for flattened ORC is set at flattenedOrc.tableName<br>
+ *  And the destination table name for nested ORC is set at nestedOrc.tableName
+ * </ul>
+ * </p>
  */
-@Getter
 @ToString
 public class ConvertibleHiveDataset extends HiveDataset {
 
-  public static final String DESTINATION_TABLE_KEY = "destination.tableName";
-  public static final String DESTINATION_DB_KEY = "destination.dbName";
-  public static final String DESTINATION_DATA_PATH_KEY = "destination.dataPath";
-  public static final String DESTINATION_CONVERSION_FORMATS_KEY = "destination.formats";
+  public static final String DESTINATION_CONVERSION_FORMATS_KEY = "destinationFormats";
+
+  // Destination formats
+  @Getter
+  private final Set<String> destFormats;
 
-  private static final String HIVE_RUNTIME_PROPERTIES_KEY_PREFIX = "hiveRuntime";
-  private final Optional<String> destinationTableName;
-  private final Optional<String> destinationDbName;
-  private final Optional<String> destinationDataPath;
-  private final Optional<HashSet<String>> formats;
-  private final Properties hiveProperties;
+  // Mapping for destination format to it's Conversion config
+  private final Map<String, ConversionConfig> destConversionConfigs;
```

Comment:
I would suggest this to be a map of Enum and ConversionConfig, and constructor resolve from destinationFormat string to enum, so that we fail fast for unsupported destination formats